### PR TITLE
Setup PlatformIO and TravisCI

### PR DESCRIPTION
**Pull request by baphled**
This PR introduces PlatformIO and TravisCI. To to help with making sure that this library compiles against various boards.

I initially had it built against 1.4, as I noticed an issues with getting 1.5 to work with one or two of my other projects.

So I figured it would be beneficial to add some automation to this project.

### CODE REVIEW

**Comment by baphled** [comment]
@nrwiersma It'd be cool to get your thoughts on this. I'm more than happy to tweak it as required :)

**Comment by nrwiersma** [comment]
Hi,

Thanks, this seems like a nice idea and something i would interested in seeing. Couple questions. Are empty folders with just READMEs really needed? Is the `main.cpp` really needed?

Thanks,
Nick

**Comment by nrwiersma** [comment]
btw I have turned Travis on for this repo, so we should see it run on the next push of this PR.

**Comment by cgmckeever** [comment]
@baphled Were there issues with the code in the new release that need to be addressed? Or were there items on your side that fixed the issue?

**Comment by baphled** [comment]
> Hi,
> 
> Thanks, this seems like a nice idea and something i would interested in seeing. Couple questions. Are empty folders with just READMEs really needed? Is the `main.cpp` really needed?
> 
> Thanks,
> Nick

@nrwiersma `mine.cpp` is just required so that we can build the library. If we remove it we won't get the library to build. Although someone else, with more knowledge of platformio may be able to correct me though.

We could move the extra classes out of `ConfigManager.h` into the `lib` directory, for example. Which will help us when it comes to writing tests later.

They're really required to get a test build running though, so more than happy to remove them.

**Comment by baphled** [comment]
> @baphled Were there issues with the code in the new release that need to be addressed? Or were there items on your side that fixed the issue?

@cgmckeever There was an issue of compiling against the ESP32 but we could resolve that later. It's easy enough to add and remove.

**Comment by baphled** [comment]
@nrwiersma I've created a POC for keeping the `lib` directory, the links below.

If you're happy with that I'll add this PR here also, once we've got this one wrapped up.

https://github.com/boodah-consulting/ConfigManager/pull/1

**Comment by nrwiersma** [comment]
I wonder if something like this might work to the same effect, but less empty changes. WDYT?

**Comment by nrwiersma** [comment]
I have been buggering around with this in a branch, seeing if we can reduce the footprint. This is what I came up with: https://github.com/nrwiersma/ConfigManager/tree/travis-test. WDYT?

I found some issues which i fixed at the same time.

**Comment by baphled** [comment]
Ah, nice, that looks good to me.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+cache:
+  directories:
+    - "~/.platformio"
+
+install:
+  - pip install -U platformio
+  - platformio update
+
+script:
+  - platformio run -e nodemcuv2 -e esp32
```
Comment (nrwiersma) [inline .travis.yml]: Use `ci` and the platform conf rather than `run`.

File: src/main.cpp [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#include "ConfigManager.h"
+
+void setup() { }
```
Comment (nrwiersma) [inline src/main.cpp]: This file isnt needed. Use the `simple` demo instead.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+
+#
+# Template #2: The project is intended to be used as a library with examples.
```
Comment (nrwiersma) [inline .travis.yml]: Clean this file up please.

File: src/main.cpp (hunk 2 above)
Comment (baphled) [reply src/main.cpp]: Oh, sweet, I didn't know you could do that :)

**Pull request merged**